### PR TITLE
Parachain node-api

### PR DESCRIPTION
**Pull request by benoitdevos**
Companion of https://github.com/logion-network/logion-collator/pull/7

* node-api is generated for parachain.
* SDK checks it does connect to Parachain.
* SS58 prefix is obtained from `api.queries.ss58Prefix`.
* Integration tests are now prefix-independent.

### Tests
* node-api tests: ✅
* client tests:  ✅
* node-api integration tests: ✅ (⚠️ currently require explicit startup/shutdown of the collator node on port 9944, for instance using zombienet as [described here](https://github.com/logion-network/logion-collator/?tab=readme-ov-file#test-locally)).
* client integration tests: ❌

### CODE REVIEW

File: packages/node-api/integration/Adapters.ts [hunk 1]
```diff
@@ -21,6 +22,7 @@ export async function toSponsorshipId() {
 
 export async function toPalletLogionLocMetadataItem() {
     const { api } = await setup();
+    const polkadotAddress = encodeAddress("5FniDvPw22DMW1TLee9N8zBjzwKXaKB2DcvZZCQU5tjmv1kb", api.queries.ss58Prefix);
```
Comment (gdethier) [inline packages/node-api/integration/Adapters.ts:25]: What about moving this logic to `getValidAccountId`? The returned object could make address comparisons etc prefix-safe.

File: packages/node-api/integration/LoAuthorityList.ts [hunk 2]
```diff
@@ -1,17 +1,20 @@
-import { ALICE, DAVE, FERDIE, setup, signAndSend } from "./Util.js";
+import { setup, signAndSend } from "./Util.js";
+import { encodeAddress } from "@polkadot/util-crypto";
 
 export async function addGuestLegalOfficer() {
     const { alice, api } = await setup();
-    const extrinsic = api.polkadot.tx.loAuthorityList.addLegalOfficer(DAVE, {
-        Guest: ALICE,
+    const dave = encodeAddress("5DAAnrj7VHTznn2AWBemMuyBwZWs6FNFjdyVXUeYum3PTXFy", api.queries.ss58Prefix);
```
Comment (gdethier) [inline packages/node-api/integration/LoAuthorityList.ts:6]: Could use `getValidAccountId` if applying my previous comment.

File: packages/node-api/integration/LoAuthorityList.ts [hunk 3]
```diff
@@ -50,6 +53,7 @@ export async function getAvailableRegions() {
 
 export async function importHost() {
     const { api, alice } = await setup();
+    const ferdie = encodeAddress("5CiPPseXPECbkjWCa6MnjNokrgYjMqmKndv2rSnekmSK2DjL", api.queries.ss58Prefix);
```
Comment (gdethier) [inline packages/node-api/integration/LoAuthorityList.ts:56]: Could use `getValidAccountId` if applying my previous comment.

**Review by gdethier — APPROVED** [review]
Good job!

File: packages/node-api/integration/LoAuthorityList.ts (hunk 2 above)
Comment (benoitdevos) [reply packages/node-api/integration/LoAuthorityList.ts]: Done in b8dee278

File: packages/node-api/integration/LoAuthorityList.ts (hunk 3 above)
Comment (benoitdevos) [reply packages/node-api/integration/LoAuthorityList.ts]: Done in b8dee278

**Pull request merged**